openquake/hazardlib/geo/surface/kite_fault.py

```diff
@@ -0,0 +1,754 @@
+    """
+
+    # Create two numpy.ndarray with the coordinates of the two profiles
+    coo1 = [(pnt.longitude, pnt.latitude, pnt.depth) for pnt in pro1.points]
+    coo2 = [(pnt.longitude, pnt.latitude, pnt.depth) for pnt in pro2.points]
+    coo1 = np.array(coo1)
+    coo2 = np.array(coo2)
+
+    # Set the profile with the smaller number of points as the first one
+    swap = 1
+    if coo2.shape[0] < coo1.shape[0]:
+        coo1, coo2 = coo2, coo1
+        swap = -1
+
+    # Process the profiles. Note that in the ideal case the two profiles
+    # require at least 5 points
+    if len(coo1) > 5 and len(coo2) > 5:
+        #
+        # create two arrays of the same lenght
+        coo1 = np.array(coo1)
+        coo2 = np.array(coo2[:coo1.shape[0]])
+        #
+        indexes = np.arange(-2, 3)
+        dff = np.zeros_like(indexes)
+        for i, shf in enumerate(indexes):
+            if shf < 0:
+                dff[i] = np.mean(abs(coo1[:shf, 2] - coo2[-shf:, 2]))
+            elif shf == 0:
+                dff[i] = np.mean(abs(coo1[:, 2] - coo2[:, 2]))
+            else:
+                dff[i] = np.mean(abs(coo1[shf:, 2] - coo2[:-shf, 2]))
+        amin = np.amin(dff)
+        res = indexes[np.amax(np.nonzero(dff == amin))] * swap
+    else:
+        d1 = np.zeros((len(coo2)-len(coo1)+1, len(coo1)))
+        d2 = np.zeros((len(coo2)-len(coo1)+1, len(coo1)))
+        for i in np.arange(0, len(coo2)-len(coo1)+1):
+            d2[i, :] = [coo2[d, 2] for d in range(i, i+len(coo1))]
+            d1[i, :] = coo1[:, 2]
+        res = np.argmin(np.sum(abs(d2-d1), axis=1))
+    return res
+
+
+def get_coords(line, idl):
+    """
+    Create a list with the coordinates of the points describing a line
+
+    :param line:
+        An instance of :class:`openquake.hazardlib.geo.line.Line`
+    :returns:
+        A list with the 3D coordinates of the line.
+    """
+    tmp = []
+    for p in line.points:
+        if p is not None:
+            if idl:
+                p.longitude = (p.longitude+360 if p.longitude < 0
+                               else p.longitude)
+            tmp.append([p.longitude, p.latitude, p.depth])
+    return tmp
+
+
+def get_mesh(pfs, rfi, sd, idl):
+    """
+    From a set of profiles creates the mesh in the forward direction from the
+    reference profile.
+
+    :param pfs:
+        List of :class:`openquake.hazardlib.geo.line.Line` instances
+    :param rfi:
+        Index of the reference profile
+    :param sd:
+        Sampling distance [km] for the edges
+    :param idl:
+        Boolean indicating the need to account for the IDL
+    :returns:
+        An updated list of the profiles i.e. a list of
+        :class:`openquake.hazardlib.geo.line.Line` instances
+    """
+    g = Geod(ellps='WGS84')
+
+    # Residual distance, last index
+    rdist = [0 for _ in range(0, len(pfs[0]))]
+    laidx = [0 for _ in range(0, len(pfs[0]))]
+
+    # New profiles
+    npr = list([copy.deepcopy(pfs[rfi])])
```

Comment:
Why the deepcopy? Please add a comment.

Reply:
Also, this function is too big, probably it can be split in a couple of helper functions.

Reply:
I know but for the time being I am focusing on adding the necessary functionalities, tests and on making sure it gives sensible results. In a second step I will improve the structure of the code.

Reply:
I added a note for the time being